scheduler/core/statscalculator/statscalculator.py

```diff
@@ -49,44 +49,56 @@ def calculate_timeline_stats(timeline: NightlyTimeline,
         scores_per_program: Dict[ProgramID, float] = {}
         programs = {}
         for night_idx in nights:
+            timeline.time_losses.setdefault(night_idx, {})
             for site in sites:
-                time_losses = {StatCalculator._FAULT_KEY: 0,
-                               StatCalculator._WEATHER_KEY: 0,
-                               StatCalculator._UNSCHEDULE_KEY: 0}
+
+                timeline_time_losses = {StatCalculator._FAULT_KEY: 0,
+                                        StatCalculator._WEATHER_KEY: 0}
+                timeline.time_losses[night_idx].setdefault(site, {})
                 # Gather unsolved interruptions during the night.
                 interruptions = []
                 for entry in timeline.timeline[night_idx][site]:
                     if isinstance(entry.event, InterruptionEvent):
                         interruptions.append(entry.event)
                     elif isinstance(entry.event, InterruptionResolutionEvent):
-                        interruptions.pop()
+                        interruptions.pop()  # remove reported interruption and register the time loss
+                        if isinstance(entry.event, FaultResolutionEvent):
+                            timeline_time_losses[StatCalculator._FAULT_KEY] += int(entry.event.time_loss.total_seconds()/60)
+                        elif isinstance(entry.event, WeatherClosureResolutionEvent):
+                            timeline_time_losses[StatCalculator._WEATHER_KEY] += int(entry.event.time_loss.total_seconds()/60)
 
+                # Unsolved interruptions for the night
                 for e in interruptions:
+                    print('event never finished:', e.description)
                     if isinstance(e, FaultEvent):
                         time_loss = timeline.timeline[night_idx][site][-1].event.time - e.time
-                        time_losses[StatCalculator._FAULT_KEY] += time_loss.total_seconds() / 60
+                        timeline_time_losses[StatCalculator._FAULT_KEY] += int(time_loss.total_seconds() / 60)
 
                     elif isinstance(e, WeatherClosureEvent):
                         time_loss = timeline.timeline[night_idx][site][-1].event.time - e.time
-                        time_losses[StatCalculator._WEATHER_KEY] += time_loss.total_seconds() / 60
+                        timeline_time_losses[StatCalculator._WEATHER_KEY] += int(time_loss.total_seconds() / 60)
```

Comment:
I guess this list may have multiple elements and it seems that time_loss is being calculated as the time between the interruption and the last night event.
This means, if you have 2 unresolved interruptions, for example, one of type FaultEvent and the other of WeatherClosureEvent, both happening at the beginning of the night, the time_loss associated to the FaultEvent will be the entire night time, and the same will happen for the WeatherClosureEvent, isn't this wrong?

By the way, I think this loop can be join with the previous one.

Reply:
As the same above if the weather closure is happening we will not have a fault but if the fault happens is for the rest of the night. Is not possible to have both at the same time. what it could happen: is fault - faultResolve - WeatherClose. Then the fault is calculated above and the weather closed is for the rest of the night.